packages/components/src/components/PipelineRun/PipelineRun.js

```diff
@@ -118,7 +118,7 @@ export /* istanbul ignore next */ class PipelineRunContainer extends Component {
           };
           taskRun = theRun; // eslint-disable-line no-param-reassign
         } else {
-          steps = stepsStatus(task.spec.steps, taskRun.status.steps);
+          steps = stepsStatus(taskRun.status.steps, taskRun.status.steps);
```

Comment:
I remember now why we used the task spec

we may not have status for all steps, so would end up leaving some steps out of the list until they begin executing
using the task spec we get the full list of steps even if we don't have status for them yet

Reply:
Hm, in what case were you seeing that happen?

When I run a Pipeline, all of the steps seem to appear even before they begin executing:
<img width="432" alt="Screen Shot 2020-01-08 at 4 05 22 PM" src="https://user-images.githubusercontent.com/16961496/72015949-c39eba80-3230-11ea-851d-ee7c7d874301.png">

Reply:
yeah I'm not seeing it anymore either... ok let's go for it 👍

Reply:
Ok, thanks @AlanGreene! The PR's ready for review now 🙂

Reply:
@ncskier looks like there was another reason for `task.spec.steps` after all, it was used to populated the step definition data displayed on the details tab

with this change we're now showing essentially the same status information (only difference being the step id on the details tab) on both the status and details tabs

I think reverting this line here and in TaskRun should resolve this issue without affecting the display of logs when no step name is given. Testing it out now... 🤞